front/migrations/db/migration_75.sql

```diff
@@ -0,0 +1,26 @@
+-- -- This migration is dependant on a backfill script
+-- -- The backfill script is: 20240906_backfill_apps_vault_ids.ts
+-- -- run psql with --set=backfilled=1 argument if you have rune the script.
+
+CREATE OR REPLACE FUNCTION perform_migration(backfilled boolean DEFAULT false)
+RETURNS VARCHAR AS $$
+BEGIN
+    IF NOT backfilled THEN
+        RAISE NOTICE 'The backfill script: 20240906_backfill_apps_vault_ids.ts is required before applying this migation. If you already did it, run psql with --set=backfilled=1 argument.';
+    END IF;
+
+-- Migration created on Sep 06, 2024
+ALTER TABLE "public"."apps" ADD COLUMN "vaultId" INTEGER REFERENCES "vaults" ("id") ON DELETE RESTRICT ON UPDATE CASCADE;
+
+    RETURN 'success';
+END;
+$$ LANGUAGE plpgsql;
+
+\if :{?backfilled}
```

Comment:
🤯  new pattern ?

Reply:
I don't get it though, if it rolls back creating the field how can you apply the migration ?

Reply:
@Fraggle added that to prevent missing a ts migration

But indeed this should only be set when I make it non null. Will fix :+1: